server/src/main/java/org/candlepin/resteasy/DateFormatter.java

```
@@ -15,34 +15,74 @@
+
+    private String[] formats;
 
     public void setAnnotations(Annotation[] annotations) {
-        DateFormat format = FindAnnotation.findAnnotation(annotations, DateFormat.class);
-        formatter = new SimpleDateFormat(format.value());
+        DateFormat annotation = FindAnnotation.findAnnotation(annotations, DateFormat.class);
+
+        this.formats = annotation.value();
     }
 
-    public Date fromString(String str) {
-        if (StringUtils.isBlank(str)) {
-            return new Date(0);
-        }
-        try {
-            return formatter.parse(str);
-        }
-        catch (ParseException e) {
-            throw new RuntimeException(e);
+    public Date fromString(String value) {
+        // If we're null/empty, we have nothing to parse, so return null
+        if (!StringUtils.isBlank(value)) {
+            // If the value is our special value for "now", return a new Date instance
+            if (DateFormat.NOW.equals(value)) {
+                return new Date();
+            }
+
+            // If we have any formats specified, use those...
+            if (this.formats != null && this.formats.length > 0) {
+                for (String format : this.formats) {
+                    try {
+                        SimpleDateFormat formatter = new SimpleDateFormat(format);
+                        return formatter.parse(value);
+                    }
+                    catch (ParseException exception) {
+                        // Whoops. Hopefully we have more formats to try...
+                    }
+                }
+
+                // If we made it here, we're out of formats to use to parse the date
+
+                // TODO: How do we get an i18n instance at this point to translate this?
+                String message = "Unable to parse date parameter: " + value;
+                throw new RuntimeException(message);
+            }
+            else {
+                try {
```

I like it as designed. If a format(s) is specified that doesn't work, fail early. I'd suggest that we beef up the error message a little, or add some logging, to describe what it attempted to do before failing.